### PR TITLE
Add an additional build configuration which will treat warnings as if…

### PR DESCRIPTION
**Pull request by nickpape-msft**
Add an additional build configuration which will treat warnings as if they are errors

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,7 @@
 *.build.log
 .DS_Store
 /common/last-install.log
+/common/npm-local
```
Comment (qinweiz) [inline .gitignore]: Those two changes do not seem to be related to this change. What are those for?

File: gulp-core-build/src/logging.ts [hunk 2]
```diff
@@ -233,7 +238,7 @@ function writeSummary(callback: () => void): void {
         }
 
         if (getWarnings().length) {
-          log('Task warnings:', gutil.colors.yellow(getWarnings().length + '\r\n' + getWarnings().join('\r\n')));
+          log('Task warnings:', gutil.colors.yellow(getWarnings().length.toString()));
```
Comment (iclanton) [inline gulp-core-build/src/logging.ts]: You shouldn't need the `toString`

Comment (nickpape-msft) [inline gulp-core-build/src/logging.ts]: Not officially, but the ts compiler is complaining otherwise. I wonder if it means the bindings for yellow() expect a string

File: .gitignore (hunk 1 above)
Comment (nickpape-msft) [inline .gitignore]: True, these are related to the version of NPMX you are using... This path was updated in the latest NPMX
